### PR TITLE
Dependency management: dep

### LINKED ISSUE #70: Setup Dependency Management
Need to introduce a dependency management tool to ensure we are keeping track of package versions. Go-modules is not supported entirely yet for many packages/libraries so in the meantime we can look at using a tool like [dep](https://github.com/golang/dep)

## Possible Solution
Many tools available. [Tools](https://github.com/golang/go/wiki/PackageManagementTools)

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment
* Version used:
* Commit hash:
* Go Version:
* Operating System and version (include distro if Linux):
* Link to your project or reproducible gist:
* I intend to submit a pull request to fix this issue: [ ]

### PR DESCRIPTION
**Pull request by dutterbutter**
Added dep to be used for dependency management until go mods are more widely supported

**[Dep Documentation](https://golang.github.io/dep/docs/introduction.html)** 

## Changes
- Added [dep](https://github.com/golang/dep) to be used for dependency management
- Initialized existing packages into `Gopkg.toml` and `Gopkg.lock`

## Tests:
```
N/A
```

`dep ensure`

> Adding a dependency

`dep ensure -add github.com/foo/bar`

> Adding a dependency with a constraint

`dep ensure -add github.com/foo/bar@1.0.0 github.com/foo/baz@master`

> Checking status of dependencies 

`dep status`

> Updating dependencies

`dep ensure -update github.com/foo/bar `

> Removing dependencies 

1. Remove the relevant `import` statements and usage in code
2. Remove the relevant `[[constraint]]` rules from `Gopkg.toml`
3. Run `dep ensure`

### Fixes:

Closes #70

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  version = "0.1.0"
+
+[[constraint]]
+  name = "github.com/libp2p/go-libp2p-peerstore"
+  version = "0.0.1"
+
+[[constraint]]
+  name = "github.com/multiformats/go-multiaddr"
+  version = "0.0.2"
+
```
Comment (noot) [inline Gopkg.toml]: do we need to include every dependency we have here? if so there are a few more we should add

Comment (dutterbutter) [reply Gopkg.toml]: these are the dependencies that currently exist in the codebase, you only add the dependency when it's being used. For the p2p dependencies, they will be included when you use `dep ensure` or `dep ensure -add` on your own branch where the dependencies are being used

Comment (noot) [reply Gopkg.toml]: i see, so on my p2p PR, i should add it to the toml file? (once this gets merged)

**Comment by priom** [comment]
@dutterbutter why are we using `dep` instead of `go modules`?

File: Gopkg.toml (hunk 1 above)
Comment (noot) [reply Gopkg.toml]: okay i see, that makes sense thanks

**Comment by dutterbutter** [comment]
@priom we will in the future but we do not know if all the packages and libraries we will need support go mods at this time

**Comment by priom** [comment]
k cool.. @dutterbutter

**Pull request merged**